client/authority-discovery/src/lib.rs

```diff
@@ -329,58 +331,67 @@ where
 		Ok(())
 	}
 
-	fn handle_dht_events(&mut self, cx: &mut Context) -> Result<()> {
+	/// Handle incoming Dht events.
+	///
+	/// Returns either:
+	///   - Poll::Pending when there are no more events to handle or
+	///   - Poll::Ready(Error) when a fatal error occurred.
+	fn handle_dht_events(&mut self, cx: &mut Context) -> Poll<Error>{
 		loop {
-			match self.dht_event_rx.poll_next_unpin(cx) {
-				Poll::Ready(Some(DhtEvent::ValueFound(v))) => {
+			match ready!(self.dht_event_rx.poll_next_unpin(cx)) {
+				Some(DhtEvent::ValueFound(v)) => {
 					if let Some(metrics) = &self.metrics {
 						metrics.dht_event_received.with_label_values(&["value_found"]).inc();
 					}
 
 					if log_enabled!(log::Level::Debug) {
 						let hashes = v.iter().map(|(hash, _value)| hash.clone());
 						debug!(
-							target: "sub-authority-discovery",
+							target: LOG_TARGET,
 							"Value for hash '{:?}' found on Dht.", hashes,
 						);
 					}
 
-					self.handle_dht_value_found_event(v)?;
+					if let Err(e) = self.handle_dht_value_found_event(v) {
+						error!(
+							target: LOG_TARGET,
+							"Failed to handle Dht value found event: {:?}", e,
+						);
+					}
 				}
-				Poll::Ready(Some(DhtEvent::ValueNotFound(hash))) => {
+				Some(DhtEvent::ValueNotFound(hash)) => {
 					if let Some(metrics) = &self.metrics {
 						metrics.dht_event_received.with_label_values(&["value_not_found"]).inc();
 					}
 
 					debug!(
-						target: "sub-authority-discovery",
+						target: LOG_TARGET,
 						"Value for hash '{:?}' not found on Dht.", hash
 					)
 				},
-				Poll::Ready(Some(DhtEvent::ValuePut(hash))) => {
+				Some(DhtEvent::ValuePut(hash)) => {
 					if let Some(metrics) = &self.metrics {
 						metrics.dht_event_received.with_label_values(&["value_put"]).inc();
 					}
 
 					debug!(
-						target: "sub-authority-discovery",
+						target: LOG_TARGET,
 						"Successfully put hash '{:?}' on Dht.", hash,
 					)
 				},
-				Poll::Ready(Some(DhtEvent::ValuePutFailed(hash))) => {
+				Some(DhtEvent::ValuePutFailed(hash)) => {
 					if let Some(metrics) = &self.metrics {
 						metrics.dht_event_received.with_label_values(&["value_put_failed"]).inc();
 					}
 
 					warn!(
-						target: "sub-authority-discovery",
+						target: LOG_TARGET,
 						"Failed to put hash '{:?}' on Dht.", hash
 					)
 				},
 				// The sender side of the dht event stream has been closed, likely due to the
 				// network terminating.
-				Poll::Ready(None) => return Err(Error::DhtEventStreamTerminated),
-				Poll::Pending => return Ok(()),
+				None => return Poll::Ready(Error::DhtEventStreamTerminated),
```

Comment:
This makes now even less sense than before :D 

We now print a warning when the node is shutting down.

I think we should just return `Ready(())` and if that happens, silently return `Ready(())` below as well.
If it is some sort of hard error, we can still print it here.